app/models/concerns/purchase_card_mixin.rb

```diff
@@ -0,0 +1,33 @@
+module PurchaseCardMixin
```

Comment:
I'd add a comment at the top saying "requires `.purchase_amount_column_name` method to be defined before `include`ing".